### DISCUSSION TITLE
Clarification on defining tasks

### DISCUSSION THREAD

**Discussion by bviebahn** [vercel/turborepo#9176] [Help] [answered]
### Summary

I'm looking at the guide for setting up Prisma:
https://turbo.build/repo/docs/guides/tools/prisma

I'm wondering about the part where the tasks are defined:
https://turbo.build/repo/docs/guides/tools/prisma#figuring-out-the-scripts

The database package has this script: `"db:generate": "prisma generate"`,
which then also has to be set up in the turbo config:
```json
{
  "tasks": {
    "dev": {
      "dependsOn": ["^db:generate"],
      "cache": false
    },
    "build": {
      "dependsOn": ["^db:generate"],
      "outputs": ["your-outputs-here"]
    },
    "db:generate": {
      "cache": false
    }
  }
}
```

I would have thought the "proper/clean" way to do this would be to just add a dev and a build script to the database package, like I've done in my other packages, and seen in many other examples:
```
"dev": "prisma generate --watch",
"build": "prisma generate",
```

If we had many packages, and each of them would choose a different name for their dev/build scripts, I guess this could quickly become a little chaotic and error prone. Of course both works, I'm just wondering if I'm missing something?

**Comment by anthonyshew** [comment, marked as answer]
You could choose to set up the tasks as you've described. The example is set up the way it is for simplicity, but is, of course, adjustable to your preferences.
